src/components/pod/pod.component.js

```diff
@@ -26,328 +25,220 @@ const marginPropTypes = filterStyledSystemMarginProps(
-      >
-        <Icon type="undo" />
-      </StyledUndoButton>
-    );
-  }
-
-  linkProps = () => {
-    const { onEdit } = this.props;
-    let props = {};
-
-    if (typeof onEdit === "string") {
-      props.to = onEdit;
-    } else if (typeof onEdit === "object") {
-      props = onEdit;
-    }
-
-    return props;
-  };
-
-  editEvents() {
-    const props = {
-      onMouseEnter: this.handleEditMouseEnter,
-      onMouseLeave: this.handleEditMouseLeave,
-      onFocus: this.handleEditFocus,
-      onBlur: this.handleEditBlur,
-    };
-
-    if (typeof this.props.onEdit === "function") {
-      props.onClick = this.processPodAction(this.props.onEdit);
-      props.onKeyDown = this.processPodAction(this.props.onEdit);
+    function undo() {
+      return (
+        <StyledUndoButton
+          onMouseEnter={() => setUndoHovered(true)}
+          onMouseLeave={() => setUndoHovered(false)}
+          onFocus={() => setUndoFocused(true)}
+          onBlur={() => setUndoFocused(false)}
+          data-element="undo-container"
+          internalEditButton={internalEditButton}
+          isFocused={isUndoFocused}
+          isHovered={isUndoHovered}
+          noBorder={!border}
+          size={size}
+          variant={variant}
+          onKeyDown={processPodAction(onUndo)}
+          onAction={processPodAction(onUndo)}
+        >
+          <Icon type="undo" />
+        </StyledUndoButton>
+      );
     }
 
-    return props;
-  }
+    function linkProps() {
+      if (typeof onEdit === "string") {
+        return { to: onEdit };
+      }
 
-  processPodAction = (action) => (ev) => {
-    if (Event.isEnterKey(ev) || !Event.isEventType(ev, "keydown")) {
-      ev.preventDefault();
-      action(ev);
-    }
-  };
-
-  shouldContentHaveEditEvents() {
-    const {
-      triggerEditOnContent,
-      displayEditButtonOnHover,
-      onEdit,
-    } = this.props;
-    return (triggerEditOnContent || displayEditButtonOnHover) && onEdit;
-  }
-
-  render() {
-    const {
-      variant,
-      border,
-      editContentFullWidth,
-      internalEditButton,
-      onEdit,
-      onUndo,
-      onDelete,
-      softDelete,
-      size,
-      title,
-      height,
-      ...rest
-    } = this.props;
-
-    const {
-      isEditFocused,
-      isEditHovered,
-      isDeleteFocused,
-      isDeleteHovered,
-    } = this.state;
-
-    let podHeight;
-
-    if (this.context.heightOfTheLongestPod) {
-      podHeight = `${this.context.heightOfTheLongestPod}px`;
-    }
-
-    if (height && typeof height === "number") {
-      podHeight = `${height}px`;
-    } else if (height) {
-      podHeight = height;
+      return onEdit;
     }
 
     return (
       <StyledPod
         {...rest}
-        className={this.props.className}
+        className={className}
         internalEditButton={internalEditButton}
-        {...tagComponent("pod", this.props)}
+        {...tagComponent("pod", rest)}
         height={podHeight}
```

Comment:
Maybe we can do just 
```js
`${String(height)px}`
```
So the conditional that is above in this file won't be needed

Reply:
as @edleeks87 already said, we could end up with `pxpx`